bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/GsonMapper.java

```diff
@@ -0,0 +1,344 @@
+                logger.warn("Cannot parse device, because firstLevelMap does not have [device] key!\n{}", str);
+            }
+            return empty();
+        }
+        var objLevel2 = firstLevelMap.get("device");
+
+        if (!(objLevel2 instanceof Map<?, ?> map)) {
+            logger.warn("Cannot parse device, because object is not type of map!\n{}", obj);
+            return empty();
+        }
+
+        // parse `dns`
+        if (!map.containsKey("dsn")) {
+            if (logger.isWarnEnabled()) {
+                var str = map.entrySet().stream().map(entry -> format("%s=%s", entry.getKey(), entry.getValue()))
+                        .collect(Collectors.joining("\n"));
+                logger.warn("Cannot parse device, because map does not have [dsn] key!\n{}", str);
+            }
+            return empty();
+        }
+        var dsn = requireNonNull((String) map.get("dsn"));
+
+        // parse `name`
+        if (!map.containsKey("product_name")) {
+            if (logger.isWarnEnabled()) {
+                var str = map.entrySet().stream().map(entry -> format("%s=%s", entry.getKey(), entry.getValue()))
+                        .collect(Collectors.joining("\n"));
+                logger.warn("Cannot parse device, because map does not have [product_name] key!\n{}", str);
+            }
+            return empty();
+        }
+        var name = requireNonNull((String) map.get("product_name"));
+
+        // parse `properties`
+        var list = map.entrySet().stream().filter(entry -> entry.getKey() != null)
+                .filter(entry -> !"name".equals(entry.getKey())).filter(entry -> !"base_type".equals(entry.getKey()))
+                .filter(entry -> !"read_only".equals(entry.getKey()))
+                .filter(entry -> !"direction".equals(entry.getKey()))
+                .filter(entry -> !"data_updated_at".equals(entry.getKey()))
+                .filter(entry -> !"product_name".equals(entry.getKey()))
+                .filter(entry -> !"display_name".equals(entry.getKey()))
+                .filter(entry -> !"value".equals(entry.getKey()))
+                .map(entry -> new Pair<>(requireNonNull(entry.getKey()).toString(), (Object) entry.getValue()))
+                .toList();
+        Map<@NotNull String, @Nullable Object> properties = new LinkedHashMap<>();
+        for (var entry : list) {
+            properties.put(entry.key, entry.value);
+        }
+        properties = Collections.unmodifiableMap(properties);
+
+        return Optional.of(new Device(dsn.trim(), name.trim(), properties));
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private <T> T tryParseBody(@Nullable String body, TypeToken<T> typeToken, T defaultValue) {
+        if (body == null) {
+            return defaultValue;
+        }
+        try {
+            return gson.fromJson(body, typeToken);
+        } catch (JsonSyntaxException e) {
+            logger.warn("Error when parsing body!\n{}", body, e);
```

Comment:
This should be changed to debug for the overall message and trace for the body details.

Reply:
done